### PR TITLE
Add table azure_eventhub_namespace. Closes #197

### LINKED ISSUE #197: Add table azure_eventhub_namespace
**References**
https://docs.microsoft.com/en-in/rest/api/eventhub/preview/namespaces/list

### PR DESCRIPTION
**Pull request by khushboo9024**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT 300

SETUP: tests/azure_eventhub_namespace []

PRETEST: tests/azure_eventhub_namespace

TEST: tests/azure_eventhub_namespace
Running terraform
data.azurerm_client_config.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
azurerm_resource_group.named_test_resource: Creating...
azurerm_resource_group.named_test_resource: Creation complete after 4s [id=/subscriptions/********-****-****-****-************/resourceGroups/turbottest81060]
azurerm_eventhub_namespace.named_test_resource: Creating...
azurerm_eventhub_namespace.named_test_resource: Still creating... [10s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [20s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [30s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [40s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [50s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [1m0s elapsed]
azurerm_eventhub_namespace.named_test_resource: Still creating... [1m10s elapsed]
azurerm_eventhub_namespace.named_test_resource: Creation complete after 1m15s [id=/subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

location = westus
resource_aka = azure:///subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060
resource_aka_lower = azure:///subscriptions/********-****-****-****-************/resourcegroups/turbottest81060/providers/microsoft.eventhub/namespaces/turbottest81060
resource_id = /subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060
resource_name = turbottest81060
subscription_id = ********-****-****-****-************

Running SQL query: test-get-query.sql
[
  {
    "id": "/subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060",
    "is_auto_inflate_enabled": false,
    "kafka_enabled": true,
    "name": "turbottest81060",
    "region": "westus",
    "resource_group": "turbottest81060",
    "type": "Microsoft.EventHub/Namespaces"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060",
      "azure:///subscriptions/********-****-****-****-************/resourcegroups/turbottest81060/providers/microsoft.eventhub/namespaces/turbottest81060"
    ],
    "name": "turbottest81060",
    "tags": {
      "name": "turbottest81060"
    },
    "title": "turbottest81060"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "id": "/subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060",
    "name": "turbottest81060"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "azure:///subscriptions/********-****-****-****-************/resourceGroups/turbottest81060/providers/Microsoft.EventHub/namespaces/turbottest81060",
      "azure:///subscriptions/********-****-****-****-************/resourcegroups/turbottest81060/providers/microsoft.eventhub/namespaces/turbottest81060"
    ],
    "name": "turbottest81060",
    "tags": {
      "name": "turbottest81060"
    },
    "title": "turbottest81060"
  }
]
✔ PASSED

POSTTEST: tests/azure_eventhub_namespace

TEARDOWN: tests/azure_eventhub_namespace

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
## Examples

### Basic info

```sql
select
  name,
  id,
  type,
  provisioning_state,
  created_at
from
  azure_eventhub_namespace;
```
```
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+--------------------+--------------------------+
| name         | id                                                                                                                           | type                          | provisioning_state | created_at               |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+--------------------+--------------------------+
| ttest6777    | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/ttest6777    | Microsoft.EventHub/Namespaces | Succeeded          | 2021-07-24T15:48:29.77Z  |
| test-without | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/test-without | Microsoft.EventHub/Namespaces | Succeeded          | 2021-07-26T05:15:40.253Z |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+--------------------+--------------------------+
```

### List Namespaces with no virtual network service endpoint configured

```sql
select
  name,
  id,
  type,
  network_rule_set -> 'properties' -> 'virtualNetworkRules' as virtual_network_rules
from
  azure_eventhub_namespace
where
  network_rule_set  -> 'properties' -> 'virtualNetworkRules' = '[]';
```
```
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+-----------------------+
| name         | id                                                                                                                           | type                          | virtual_network_rules |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+-----------------------+
| test-without | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/test-without | Microsoft.EventHub/Namespaces | []                    |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+-----------------------
```

### List Namespaces with encryption disabled

```sql
select
  name,
  id,
  type,
  encryption
from
  azure_eventhub_namespace
where
  encryption is null;
```
```
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+------------+
| name         | id                                                                                                                           | type                          | encryption |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+------------+
| test-without | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/test-without | Microsoft.EventHub/Namespaces | <null>     |
| ttest6777    | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/ttest6777    | Microsoft.EventHub/Namespaces | <null>     |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------------+------------+
```

### List Namespaces with auto_inflate disabled

```sql
select
  name,
  id,
  is_auto_inflate_enabled,
  type
from
  azure_eventhub_namespace
where
  not is_auto_inflate_enabled;
```
```
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------+-------------------------------+
| name         | id                                                                                                                           | is_auto_inflate_enabled | type                          |
+--------------+------------------------------------------------------------------------------------------------------------------------------+-------------------------+-------------------------------+
| test-without | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/test-without | false                   | Microsoft.EventHub/Namespaces |
| ttest6777    | /subscriptions/********-****-****-****-************/resourceGroups/demo/providers/Microsoft.EventHub/namespaces/ttest6777    | false                   | Microsoft.EventHub/Namespaces |
```
</details>

### CODE REVIEW

File: azure/table_azure_eventhub_namespace.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+	"context"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/eventhub/mgmt/2018-01-01-preview/eventhub"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION ////
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:13]:
```suggestion
//// TABLE DEFINITION
```

File: azure/table_azure_eventhub_namespace.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+func tableAzureEventHubNamespace(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_eventhub_namespace",
+		Description: "Azure Event Hub Namespace",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"name", "resource_group"}),
+			Hydrate:           getAzureEventHubNamespace,
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:21]:
```suggestion
			Hydrate:           getEventHubNamespace,
```

File: azure/table_azure_eventhub_namespace.go [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceGroupNotFound", "ResourceNotFound", "404"}),
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listAzureEventHubNamespaces,
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:25]:
```suggestion
			Hydrate: listEventHubNamespaces,
```

File: azure/table_azure_eventhub_namespace.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+			},
+			{
+				Name:        "id",
+				Description: "The resource Id.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "type",
+				Description: "The Resource type.",
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:41]:
```suggestion
				Description: "The resource type.",
```

File: azure/table_azure_eventhub_namespace.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+		},
+		Columns: []*plugin.Column{
+			{
+				Name:        "name",
+				Description: "The name of the resource.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "id",
+				Description: "The resource Id.",
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:35]:
```suggestion
				Description: "The ID of the resource.",
```

File: azure/table_azure_eventhub_namespace.go [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+				Transform:   transform.FromGo(),
+			},
+			{
+				Name:        "type",
+				Description: "The Resource type.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "provisioning_state",
+				Description: "Provisioning state of the Namespace.",
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:46]:
```suggestion
				Description: "Provisioning state of the namespace.",
```

File: azure/table_azure_eventhub_namespace.go [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+			},
+			{
+				Name:        "cluster_arm_id",
+				Description: "Cluster ARM ID of the Namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("EHNamespaceProperties.ClusterArmId"),
+			},
+			{
+				Name:        "is_auto_inflate_enabled",
+				Description: "Value that indicates whether AutoInflate is enabled for eventhub namespace.",
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:64]:
```suggestion
				Description: "Indicates whether AutoInflate is enabled for eventhub namespace, or not.",
```

File: azure/table_azure_eventhub_namespace.go [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+				Name:        "subscription_id",
+				Description: ColumnDescriptionSubscription,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ID").Transform(idToSubscriptionID),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION ////
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:177]:
```suggestion
//// LIST FUNCTION
```

File: azure/table_azure_eventhub_namespace.go [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+			d.StreamListItem(ctx, namespace)
+		}
+		result.NextWithContext(context.Background())
+		pagesLeft = result.NotDone()
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS ////
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:207]:
```suggestion
//// HYDRATE FUNCTIONS
```

File: azure/table_azure_eventhub_namespace.go [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+
+func getAzureEventHubNamespace(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getAzureComputeSnapshot")
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:210]:
```suggestion
	plugin.Logger(ctx).Trace("getEventHubNamespace")
```

File: azure/table_azure_eventhub_namespace.go [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+
+	name := d.KeyColumnQuals["name"].GetStringValue()
+	resourceGroup := d.KeyColumnQuals["resource_group"].GetStringValue()
+
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:214]: Add check for empty inputs

File: docs/tables/azure_eventhub_namespace.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+  id,
+  type,
+  provisioning_state,
+  created_at
+from
+  azure_eventhub_namespace;
+```
+
+
+### List Namespaces with no virtual network service endpoint configured
```
Comment (Subhajit97) [inline docs/tables/azure_eventhub_namespace.md:21]:
```suggestion
### List Namespaces not configured to use virtual network service endpoint
```

File: docs/tables/azure_eventhub_namespace.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+  type,
+  network_rule_set -> 'properties' -> 'virtualNetworkRules' as virtual_network_rules
+from
+  azure_eventhub_namespace
+where
+  network_rule_set -> 'properties' -> 'virtualNetworkRules' = '[]';
+```
+
+
+### List Namespaces with encryption disabled
```
Comment (Subhajit97) [inline docs/tables/azure_eventhub_namespace.md:36]:
```suggestion
### List unencrypted namespaces
```

File: azure/table_azure_eventhub_namespace.go [hunk 14]
```diff
@@ -0,0 +1,258 @@
...
+//// TABLE DEFINITION ////
+
+func tableAzureEventHubNamespace(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "azure_eventhub_namespace",
+		Description: "Azure Event Hub Namespace",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.AllColumns([]string{"name", "resource_group"}),
+			Hydrate:           getAzureEventHubNamespace,
+			ShouldIgnoreError: isNotFoundError([]string{"ResourceGroupNotFound", "ResourceNotFound", "404"}),
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:22]: @khushboo9024 Can you also validate the error returned, when invalid input is provided?

File: azure-test/tests/azure_eventhub_namespace/test-turbot-query.sql [hunk 15]
```diff
@@ -0,0 +1,3 @@
+select name, akas, title, tags
+from azure.azure_eventhub_namespace
+where name = '{{resourceName}}' and resource_group = '{{resourceName}}';
```
Comment (Subhajit97) [inline azure-test/tests/azure_eventhub_namespace/test-turbot-query.sql:3]:
```suggestion
where name = '{{ resourceName }}' and resource_group = '{{ resourceName }}';
```

File: azure/table_azure_eventhub_namespace.go [hunk 16]
```diff
@@ -0,0 +1,263 @@
...
+				Name:        "provisioning_state",
+				Description: "Provisioning state of the namespace.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("EHNamespaceProperties.ProvisioningState"),
+			},
+			{
+				Name:        "created_at",
+				Description: "The time the Namespace was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("EHNamespaceProperties.CreatedAt"),
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:54]:
```suggestion
				Transform:   transform.FromField("EHNamespaceProperties.CreatedAt").Transform(convertDateToTime),
```

File: azure/table_azure_eventhub_namespace.go [hunk 17]
```diff
@@ -0,0 +1,270 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ID").Transform(idToAkas),
+			},
+
+			// Azure standard column
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Location").Transform(toLower),
```
Comment (Subhajit97) [inline azure/table_azure_eventhub_namespace.go:172]:
```suggestion
				Transform:   transform.FromField("Location").Transform(formatRegion).Transform(toLower),
```

File: docs/tables/azure_eventhub_namespace.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+  type,
+  encryption
+from
+  azure_eventhub_namespace
+where
+  encryption is null;
+```
+
+
+### List namespaces with auto inflate disabled
```
Comment (Subhajit97) [inline docs/tables/azure_eventhub_namespace.md:51]:
```suggestion
### List namespaces with auto-inflate disabled
```

File: azure-test/tests/azure_eventhub_namespace/test-get-expected.json [hunk 19]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "id": "{{ output.resource_id.value }}",
+    "is_auto_inflate_enabled": false,
+    "kafka_enabled": true,
+    "name": "{{ resourceName }}",
+    "region": "west us",
```
Comment (Subhajit97) [inline azure-test/tests/azure_eventhub_namespace/test-get-expected.json:7]: Do not use hardcoded values

**Review by Subhajit97 — APPROVED** [review]

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**